### PR TITLE
[4537] Updated the message on the irods::error constructor if a new m…

### PR DESCRIPTION
**Pull request by JustinKyleJames**
…essage is passed in.

### CODE REVIEW

File: lib/core/src/irods_error.cpp [hunk 1]
```diff
@@ -136,6 +136,11 @@ const char *error::status_token_     = " status [";
         std::string  _fcn,
         const error& _rhs ) :
         error(_rhs) {
+ 
+        if (_msg != "") {
```
Comment (korydraughn) [inline lib/core/src/irods_error.cpp]: Is this statement checking for emptiness? If so, the **empty** member function should be used. 
```
if (!_msg.empty()) {
    ...
}
```

File: lib/core/src/irods_error.cpp [hunk 2]
```diff
@@ -136,6 +136,11 @@ const char *error::status_token_     = " status [";
...
+            message_ = _msg;
```
Comment (korydraughn) [inline lib/core/src/irods_error.cpp]: You can possibly use `std::move` here given that `_msg` arrives as a copy.

Comment (JustinKyleJames) [reply lib/core/src/irods_error.cpp]: Could not use std::move since _msg is being used below this point.

**Comment by trel** [comment]
looks good - please #.

**Comment by JustinKyleJames** [comment]
\# added

**Pull request merged**